codechecker_common/plist_parser.py

```diff
@@ -51,6 +52,70 @@
 LOG = get_logger('report')
 
 
+class LXMLPlistEventHandler(object):
+    """
+    Basic lxml event handler.
+    """
+    def start(self, tag, attrib):
+        pass
+
+    def end(self, tag):
+        pass
+
+    def data(self, data):
+        pass
+
+    def comment(self, text):
+        pass
+
+    def close(self):
+        return "closed!"
+
+
+class LXMLPlistParser(PlistParser):
+    """
+    Plist parser which uses the lxml library to parse XML data.
+
+    The benefit of this library that this is faster than other libraries so it
+    will improve the performance of the plist parsing.
+    """
+    def __init__(self):
+        PlistParser.__init__(self)
+
+        self.event_handler = LXMLPlistEventHandler()
+        self.event_handler.start = self.handleBeginElement
+        self.event_handler.end = self.handleEndElement
+        self.event_handler.data = self.handleData
+
+        from lxml.etree import XMLParser
+        self.parser = XMLParser(target=self.event_handler)
+
+    def parse(self, fileobj):
+        from lxml.etree import parse
+        parse(fileobj, self.parser)
+        return self.root
+
+
+def read_plist(path):
+    """
+    Read a .plist file. Return the unpacked root object (which usually is a
+    dictionary).
+
+    Use 'lxml' library to read the given plist file if it is available,
+    otherwise use 'plistlib' library.
+    """
+    try:
+        imp.find_module('lxml')
+
+        with open(path) as data:
```

Comment:
Using a simple open will try to decode the content of the file automatically.
I would prefer to use io.open or read it in binary before parsing the data by the xml parser.